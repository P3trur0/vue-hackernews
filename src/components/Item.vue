<template>
  <li class="news-item">
    <a :href="item.url">{{ item.title }}</a>
      {{ item.by }} {{ item.url }} {{ item.score }}
  </li>
</template>

<script>
export default {
  name: 'news-item',
  props: ['item']
}
</script>
