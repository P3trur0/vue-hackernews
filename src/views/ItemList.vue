<template>
  <item />
</template>

<script>
import Item from '../components/Item.vue'

export default {
  name: 'app',
  components: {
    Item
  }
}
</script>
