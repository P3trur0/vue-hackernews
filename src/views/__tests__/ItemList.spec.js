import { shallow } from 'vue-test-utils'
import ItemList from '../ItemList.vue'
import Item from '../../components/Item.vue'

describe('ItemList.vue', () => {
  test('renders an Item for each item in window.items', () => {
    window.items = [{}, {}, {}]
    const wrapper = shallow(ItemList)
    expect(wrapper.findAll(Item).length).toEqual(window.items.length)
  })

  test('passes an item object to each Item component', () => {
    window.items = [{}, {}, {}]
    const wrapper = shallow(ItemList)
    const ItemsArray = wrapper.findAll(Item)
    ItemsArray.wrappers.forEach((wrapper, i) => {
      expect(wrapper.vm.item).toBe(window.items[i])
    })
  })

  test('passes an item object to each Item component', () => {
    window.items = [{}, {}, {}]
    const wrapper = shallow(ItemList)
    const itemsArray = wrapper.findAll(Item)
    itemsArray.wrappers.forEach((wrapper, i) => {
      expect(wrapper.props().item).toBe(window.items[i])
    })
  })
})
