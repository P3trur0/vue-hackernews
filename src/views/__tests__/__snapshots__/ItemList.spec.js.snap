// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList.vue renders correctly 1`] = `
<div class="news-view">
    <div class="news-list-nav">
        <a class="disabled">&lt; prev</a> <span>1/2</span>
        <router-link to="/top/2">more &gt;</router-link>
    </div>
    <div class="news-list">
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
    </div>
</div>
`;
